@JackBoosY Thanks for your changes!
I have tried to set up the RecastDemo project from the recastnavigation/recastnavigation repository. So far I have noticed two things:

First:
`#include` directives (for example to: `#include <recast/Recast/Recast.h>`) only work if there is at least one other vcpkg library linked to the RecastDemo target.
So this:
`target_link_libraries(RecastDemo PRIVATE recast [...])` would result in `fatal error C1083: Cannot open include file: 'recast/Recast/Recast.h': No such file or directory`

While this (linking completely unrelated vcpkg library `nlohmann_json::nlohmann_json` ):
`target_link_libraries(RecastDemo PRIVATE recast nlohmann_json::nlohmann_json  [...])` does not produce said error.

----

Second:
Upon further investigation it seem that certain `#include` directives in the library itself don't work correctly.
This leads to build errors like: `[...]\vcpkg\installed\x64-windows\include\recast\DetourCrowd\DetourPathCorridor.h(22): fatal error C1083: Cannot open include file: 'DetourNavMeshQuery.h': No such file or directory`

`#include "DetourNavMeshQuery.h"` is expected to be in the same folder as `DetourPathCorridor.h`, but they are both in different sub-folders. Namely:

- `[...]\vcpkg\installed\x64-windows\include\recast\DetourCrowd` and
- `[...]\vcpkg\installed\x64-windows\include\recast\Detour`

----

I hope this was helpful. If there's anything I can help with further, feel free to let me know.

--- a/ports/recast/CMakeLists.txt
+++ b/ports/recast/CMakeLists.txt
@@ -42,6 +42,11 @@ target_include_directories(
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/DetourTileCache/Include>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Recast/Include>
   $<INSTALL_INTERFACE:include>
+  $<INSTALL_INTERFACE:include/Detour/Include>
+  $<INSTALL_INTERFACE:include/DetourCrowd/Include>
+  $<INSTALL_INTERFACE:include/DetourTileCache/Include>
+  $<INSTALL_INTERFACE:include/Recast/Include>
+  $<INSTALL_INTERFACE:include/DebugUtils/Include>
 )
 
 install(

--- a/ports/recast/CMakeLists.txt
+++ b/ports/recast/CMakeLists.txt
@@ -1,14 +1,6 @@
 cmake_minimum_required(VERSION 3.8)
 project(recast)
 
-include_directories(
-  DebugUtils/Include
-  Detour/Include
-  DetourCrowd/Include
-  DetourTileCache/Include
-  Recast/Include
-)
-
 set(SRC
   DebugUtils/Source/DebugDraw.cpp
   DebugUtils/Source/DetourDebugDraw.cpp
@@ -41,6 +33,16 @@ set(SRC
 )
 
 add_library(recast ${SRC})
+
+target_include_directories(
+  recast PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/DebugUtils/Include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Detour/Include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/DetourCrowd/Include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/DetourTileCache/Include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Recast/Include>
+  $<INSTALL_INTERFACE:include>
+)
 
 install(
   TARGETS recast

--- a/versions/r-/recast.json
+++ b/versions/r-/recast.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "949b280dbfefbc4ad6c1af2141e2632786baa30d",
+      "git-tree": "2f34af0b398b4c37f0cdf3fcd8dc9db965e4dd5e",
       "version": "1.5.1",
       "port-version": 4
     },

